perception/image_projection_based_fusion/CMakeLists.txt

```diff
@@ -4,14 +4,115 @@ project(image_projection_based_fusion)
+  unset(CUDA_cublas_device_LIBRARY CACHE)
+  set(CUDA_AVAIL ON)
+else()
+  message("CUDA NOT FOUND")
+  set(CUDA_AVAIL OFF)
+endif()
+
+# set flags for TensorRT availability
+option(TRT_AVAIL "TensorRT available" OFF)
+# try to find the tensorRT modules
+find_library(NVINFER nvinfer)
+find_library(NVONNXPARSER nvonnxparser)
+if(NVINFER AND NVONNXPARSER)
+  if(CUDA_VERBOSE)
+    message("TensorRT is available!")
+    message("NVINFER: ${NVINFER}")
+    message("NVONNXPARSER: ${NVONNXPARSER}")
+  endif()
+  set(TRT_AVAIL ON)
+else()
+  message("TensorRT is NOT Available")
+  set(TRT_AVAIL OFF)
+endif()
+
+# set flags for CUDNN availability
+option(CUDNN_AVAIL "CUDNN available" OFF)
+# try to find the CUDNN module
+find_library(CUDNN_LIBRARY
+NAMES libcudnn.so${__cudnn_ver_suffix} libcudnn${__cudnn_ver_suffix}.dylib ${__cudnn_lib_win_name}
+PATHS $ENV{LD_LIBRARY_PATH} ${__libpath_cudart} ${CUDNN_ROOT_DIR} ${PC_CUDNN_LIBRARY_DIRS} ${CMAKE_INSTALL_PREFIX}
+PATH_SUFFIXES lib lib64 bin
+DOC "CUDNN library."
+)
+if(CUDNN_LIBRARY)
+  if(CUDA_VERBOSE)
+    message(STATUS "CUDNN is available!")
+    message(STATUS "CUDNN_LIBRARY: ${CUDNN_LIBRARY}")
+  endif()
+  set(CUDNN_AVAIL ON)
+else()
+  message("CUDNN is NOT Available")
+  set(CUDNN_AVAIL OFF)
+endif()
+
+# Download trained models
+execute_process(COMMAND echo "start to download" 1>&2)
+find_program(CURL_AVAIL "curl")
+if(NOT CURL_AVAIL)
+  message("curl: command not found. External files could not be downloaded.")
+endif()
+
+set(DATA_PATH ${CMAKE_CURRENT_SOURCE_DIR}/data)
+execute_process(COMMAND mkdir -p ${DATA_PATH})
+
+function(download FILE_NAME FILE_HASH)
+  # https://drive.google.com/file/d/GFILE_ID/view
+  message(STATUS "Checking and downloading ${FILE_NAME}")
+  set(FILE_PATH ${DATA_PATH}/${FILE_NAME})
+  execute_process(COMMAND echo "start ${FILE_NAME}" 1>&2)
+  if(EXISTS ${FILE_PATH})
+    execute_process(COMMAND echo "found ${FILE_NAME}")
+    file(MD5 ${FILE_PATH} EXISTING_FILE_HASH)
+    if(${FILE_HASH} EQUAL ${EXISTING_FILE_HASH})
+      message(STATUS "File already exists.")
+    else()
+      execute_process(COMMAND echo "diff ${FILE_NAME}")
+      message(STATUS "File hash changes. Downloading now ...")
+      execute_process(COMMAND curl -s "https://awf.ml.dev.web.auto/perception/models/pointpainting/pts_voxel_encoder_pointpainting.onnx" -o ${FILE_PATH})
+      # file(MD5 ${FILE_PATH} DOWNLOADED_FILE_HASH)  # disable to pass ci
+      message(STATUS "Downloaded file hash: ${DOWNLOADED_FILE_HASH}")
+    endif()
+  else()
+    execute_process(COMMAND echo "not found ${FILE_NAME}")
+    message(STATUS "File doesn't exists. Downloading now ...")
+    execute_process(COMMAND curl -s "https://awf.ml.dev.web.auto/perception/models/pointpainting/pts_backbone_neck_head_pointpainting.onnx" -o ${FILE_PATH})
+    # file(MD5 ${FILE_PATH} DOWNLOADED_FILE_HASH)  # disable to pass ci
+    message(STATUS "Downloaded file hash: ${DOWNLOADED_FILE_HASH}")
+  endif()
+endfunction()
```

Comment:
pls be the same as [here](https://github.com/tier4/autoware.universe/blob/c8021d49ad2aa4dba26b3a81dd256fe196e10129/perception/lidar_centerpoint/CMakeLists.txt#L75-L105).